engine/table/src/main/java/io/deephaven/engine/table/impl/select/PatternFindFilter.java

```diff
@@ -0,0 +1,209 @@
+        return Collections.emptyList();
+    }
+
+    @Override
+    public WhereFilter copy() {
+        return new PatternFindFilter(columnName, pattern, invertPattern, matchNulls);
+    }
+
+    /**
+     * Creates a new filter that is the logical inversion of {@code this}.
+     *
+     * <p>
+     * Equivalent to {@code new PatternFindFilter(columnName, pattern, !invertPattern, !matchNulls)}.
+     *
+     * @return the inverted filter
+     */
+    public PatternFindFilter invert() {
+        return new PatternFindFilter(columnName, pattern, !invertPattern, !matchNulls);
+    }
+
+    @Override
+    public void filter(ObjectChunk<String, ? extends Values> values, LongChunk<OrderedRowKeys> keys,
+            WritableLongChunk<OrderedRowKeys> results) {
+        results.setSize(0);
+        for (int ix = 0; ix < values.size(); ++ix) {
+            if (matches(values.get(ix))) {
+                results.add(keys.get(ix));
+            }
+        }
+    }
+
+    private boolean matches(String value) {
+        return (matchNulls && value == null) || (value != null && (invertPattern ^ pattern.matcher(value).find()));
+    }
+
+    @Override
+    public String toString() {
+        return "PatternFindFilter{" +
+                "columnName=" + columnName +
+                ", pattern=" + pattern + " " + pattern.flags() +
+                ", invertPattern=" + invertPattern +
+                ", matchNulls=" + matchNulls +
+                '}';
+    }
+
+    private static String constructRegex(String[] values, MatchType matchType, boolean internalDisjunctive,
+            boolean removeQuotes, String columnName) {
+        if (values == null || values.length == 0) {
+            throw new IllegalArgumentException(
+                    "StringContainsFilter must be created with at least one value parameter");
+        }
+        final MatchFilter.ColumnTypeConvertor converter = removeQuotes
+                ? MatchFilter.ColumnTypeConvertorFactory.getConvertor(String.class, columnName)
+                : null;
+        final String regex;
+        final Stream<String> valueStream = Arrays.stream(values)
+                .map(val -> {
+                    if (StringUtils.isNullOrEmpty(val)) {
+                        throw new IllegalArgumentException(
+                                "Parameters to StringContainsFilter must not be null or empty");
+                    }
+                    return Pattern.quote(converter == null ? val : converter.convertStringLiteral(val).toString());
+                });
+
+        // If the match is simple, includes -any- or includes -none- we can just use a simple
+        // regex of or'd values
+        if ((matchType == MatchType.Regular && internalDisjunctive) ||
+                (matchType == MatchType.Inverted && !internalDisjunctive)) {
+            regex = valueStream.collect(Collectors.joining("|"));
+        } else {
+            // Note that internalDisjunctive is -always- false here.
```

Comment:
Can we try to understand why this comment holds?

Reply:
I believe it's because `StringContainsFilter` is really only meant for a WhereFilterFactory parsers of the form:

```
<ColumnName> [icase] [not] includes [any|all]<"String">
```

and was the only "external" caller setting `internalDisjunctive` (and testing).

I think it makes sense to move this logic explicitly to WhereFilterFactory to declutter the PatternFilter.

Reply:
I agree. I still think a better comment, or just an assertion, would improve the code.